### PR TITLE
SLEP019: Governance Update - Recognizing Contributions Beyond Code (version II)

### PR DESCRIPTION
**Pull request by cmarmo**
Dear all,
this pull request offers modifications to the SLEP019 draft, following the discussion in  #74.
The rendering is available [here](https://scikit-learn-enhancement-proposals--81.org.readthedocs.build/en/81/slep019/proposal.html).

### CODE REVIEW

File: slep019/proposal.rst [hunk 1]
```diff
@@ -85,28 +101,10 @@ Rename "Technical Committee" to "Steering Committee".
...
-A Contributor is promoted to a Core Contributor after being proposed by
-at least one existing Core Contributor. The proposal must specify which
-Core Team the Contributor will be part of. The promotion is effective
-after a vote on the private Core Contributor mailing list which must
-last for two weeks and which must reach at least two-thirds positive
-majority of the cast votes.
+The current members of the Technical Committee [8]_ will ensure the transition for a
+two-years period, starting from the acceptance of this SLEP.
+During the transition period roles which qualified as "Core Contributors" are expected to
+be represented in the Steering Committee by al least one member.
```
Comment (cmarmo) [inline slep019/proposal.rst:107]: Perhaps more details are necessary here?

File: slep019/proposal.rst [hunk 2]
```diff
@@ -185,15 +183,19 @@ This document has been placed in the public domain [9]_.
...
-   Permissions for each role, Repository roles for an organization, GitHub
-   Docs:
-   https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role
+   SLEP019: Governance Update - Recognizing Contributions Beyond Code
+   https://github.com/scikit-learn/enhancement_proposals/pull/74
 
 .. [8]
 
+   scikit-learn Techical Committee
+   https://scikit-learn.org/1.1/governance.html#technical-committee
```
Comment (cmarmo) [inline slep019/proposal.rst:192]: The Technical committee page should be updated: I'm opening an issue, I believe it should be updated for 1.2 (and the link here updated to 1.2 then)

Comment (jjerphan) [reply slep019/proposal.rst:192]: Otherwise, it can be:
```suggestion
   scikit-learn Techical Committee
   https://scikit-learn.org/stable/governance.html#technical-committee
```

File: slep019/proposal.rst [hunk 3]
```diff
@@ -185,15 +183,19 @@ This document has been placed in the public domain [9]_.
...
+
+.. [9]
+
    Decision Making Process, scikit-learn Governance and Decision-Making:
    https://scikit-learn.org/dev/governance.html#decision-making-process
```
Comment (jjerphan) [inline slep019/proposal.rst:197]: We might also preferably change this to point to the stable version of the documentation.
```suggestion
   https://scikit-learn.org/stable/governance.html#decision-making-process
```

File: slep019/proposal.rst [hunk 4]
```diff
@@ -70,13 +74,25 @@ modifications over time.
 
 Some of the proposed modification have been discussed in the monthly
 meetings, on April 25th 2022 [5]_ and September 5th 2022 [6]_.
+The current version is based on the discussion on the draft version [7]_.
```
Comment (jjerphan) [inline slep019/proposal.rst:77]: I do not think this is necessary to have this mention in the final version of the SLEP, but we might want to references all the PRs used for iterating on writing this document.

What do you think?

File: slep019/proposal.rst [hunk 5]
```diff
@@ -4,7 +4,8 @@
  SLEP019: Governance Update - Recognizing Contributions Beyond Code
 ####################################################################
 
-:Author: Julien Jerphanion <git@jjerphan.xyz>, Gaël Varoquaux <gael.varoquaux@normalesup.org>
+:Author: Julien Jerphanion <git@jjerphan.xyz>, Gaël Varoquaux <gael.varoquaux@normalesup.org>,
+         Chiara Marmo <marmochiaskl@gmail.com>
```
Comment (jjerphan) [inline slep019/proposal.rst:8]: I think this is correctly parsed and reads better. What do you think?
```suggestion
:Author: Julien Jerphanion <git@jjerphan.xyz>,
         Gaël Varoquaux <gael.varoquaux@normalesup.org>,
         Chiara Marmo <marmochiaskl@gmail.com>
```

File: slep019/proposal.rst [hunk 6]
```diff
@@ -13,9 +14,12 @@
  Abstract
 **********
 
-This SLEP proposes updating the Governance to broaden the notion of
-contribution in scikit-learn and to ease subsequent related changes to
-the Governance without requiring SLEPs.
+This SLEP proposes updating the Governance to broaden the notion of core
+contribution in scikit-learn.
```
Comment (jjerphan) [inline slep019/proposal.rst:18]: "Core contributions" implies that there are "Non-core contributions" ; is this what we intend exactly? Don't we want to recognized all contributions, and hence all contributors, including "Core contributors" i.e. contributors that are notably involved in the project?
```suggestion
This SLEP proposes updating the Governance to broaden the notion of contribution in scikit-learn.
```

File: slep019/proposal.rst [hunk 7]
```diff
@@ -13,9 +14,12 @@
...
+At this aim, the Technical Committee is renamed as Steering Committee, opening the
+decision process to new to be defined core roles.
+New core roles are allowed to be defined without requiring new SLEPs, to ease subsequent
+related changes to the Governance.
```
Comment (jjerphan) [inline slep019/proposal.rst:22]: :100:

File: slep019/proposal.rst [hunk 8]
```diff
@@ -70,13 +74,25 @@ modifications over time.
...
 ===================================
 
 Explicitly define Contributions and emphasize the importance of non-code
 contributions in the Governance structure.
 
+New roles and core roles, and susequent teams will be defined.
+Core Team members will become scikit-learn "Core Contributors":
+a Contributor will be promoted to a Core Contributor after being proposed by
+at least one existing Core Contributor. The proposal must specify which
+Core Team the Contributor will be part of. The promotion will be effective
```
Comment (jjerphan) [inline slep019/proposal.rst:89]: This mentions "Core Team". Should we first define what are Core Teams (and enumerate them)?

File: slep019/proposal.rst (hunk 1 above)
Comment (jjerphan) [reply slep019/proposal.rst:107]: I think so. Probably we should give a grasp about eventual roles in this SLEP without have them be exhaustively defined. What do you think?

Also, here are just some minor rephrasing suggestions:
```suggestion
The current members of the Technical Committee [8]_ will ensure the transition
for a two-years period, starting from the acceptance of this SLEP.
During the transition period, each role among "Core Contributors" is expected
to be represented in the Steering Committee by at least one member having this
role.
```

**Review by jjerphan — COMMENTED** [review]
Thank you for moving this forward, @cmarmo!

Here are some comments.

File: slep019/proposal.rst (hunk 4 above)
Comment (cmarmo) [reply slep019/proposal.rst:77]: Yes, I would like to reference somewhere all the PRs, so people could see this is not a personal effort of the people listed as authors, but a mediation of all the ideas that came up during the discussion.

File: slep019/proposal.rst (hunk 8 above)
Comment (cmarmo) [reply slep019/proposal.rst:89]: As per https://github.com/scikit-learn/enhancement_proposals/pull/74#issuecomment-1310802614, let accept here the general principle of new core roles and define a simple way to create them.
Once this SLEP accepted it will be easier and straightforward to introduce those new roles.

File: slep019/proposal.rst (hunk 2 above)
Comment (cmarmo) [reply slep019/proposal.rst:192]: I want to give a snapshot of the TC at the time of the acceptance of the SLEP.
This will probably become https://scikit-learn.org/1.2/governance.html#technical-committee but the link is not existent for now.

File: slep019/proposal.rst (hunk 1 above)
Comment (cmarmo) [reply slep019/proposal.rst]: > Probably we should give a grasp about eventual roles in this SLEP without have them be exhaustively defined. What do you think?

See https://github.com/scikit-learn/enhancement_proposals/pull/74#issuecomment-1310802614

I was referring to details about the nomination/election of new members.

File: slep019/proposal.rst [hunk 9]
```diff
@@ -85,28 +101,11 @@ Rename "Technical Committee" to "Steering Committee".
...
-   -  Development Team
-
-A Contributor is promoted to a Core Contributor after being proposed by
-at least one existing Core Contributor. The proposal must specify which
-Core Team the Contributor will be part of. The promotion is effective
-after a vote on the private Core Contributor mailing list which must
-last for two weeks and which must reach at least two-thirds positive
-majority of the cast votes.
+The current members of the Technical Committee [8]_ will ensure the transition
+for a two-years period, starting from the acceptance of this SLEP.
```
Comment (betatim) [inline slep019/proposal.rst:105]: Why two years? Is it a starting point to be discussed or is there some thinking behind why two is a good number?

My worry and experience with other governance changes/refactors/restructuring is that "work expands to take the time allocated to it". For me four weeks is (obviously) too short, and equally two years is (obviously) too long. At least it isn't clear to me that there is so much work needed that people would react with "two years to get this done, we better start right now and consistently work towards it because we might miss the deadline otherwise."

How about six months?

For me overly long deadlines are a signal that "this isn't really that important" as the first reaction by people having to deliver on the deadline will be procrastination and not action A deadline has to create a sense of urgency, not panic, but urgency. Otherwise it is IMHO useless to have a deadline.

Comment (cmarmo) [reply slep019/proposal.rst:105]: > Why two years?

Some experience with decision times in scikit-learn.... :innocent: 
But I am open to optimism, I believe a renewal of the SC is doable in one year: I am scared by the frustration that a deadline of 6 months could generate ... at least for me.
Quick reactions about this PR from all the core-devs might eventually push me to further shorten the deadline... :pray:

Comment (cmarmo) [reply slep019/proposal.rst:105]: > Is it a starting point to be discussed

Yes.

File: slep019/proposal.rst (hunk 4 above)
Comment (cmarmo) [reply slep019/proposal.rst]: I put both PRs in the ref. @jjerphan , let me know if you are ok with this formulation.

File: slep019/proposal.rst (hunk 9 above)
Comment (cmarmo) [reply slep019/proposal.rst]: Reduced the transition period to one year... :crossed_fingers: :

Comment (betatim) [reply slep019/proposal.rst]: I (still) believe that people deliver when you have high expectations. More practically, I expect people who agreed to serve in some role for a project (be that maintainer, team lead, club president, steering committee, etc) do so knowing that it will require commitment. It isn't like a honorary title that a university gives out or an award where you don't have to do anything in return for receiving it.

(I also get disappointed because that doesn't happen, but I remain optimistic :D)

File: slep019/proposal.rst (hunk 4 above)
Comment (jjerphan) [reply slep019/proposal.rst]: LGTM!

File: slep019/proposal.rst (hunk 8 above)
Comment (jjerphan) [reply slep019/proposal.rst:89]: This seems fair and reasonable to me.

File: slep019/proposal.rst (hunk 2 above)
Comment (jjerphan) [reply slep019/proposal.rst:198]: It now is!

```suggestion
   scikit-learn Techical Committee
   https://scikit-learn.org/1.2/governance.html#technical-committee
```

File: slep019/proposal.rst [hunk 10]
```diff
@@ -13,9 +15,11 @@
  Abstract
 **********
 
-This SLEP proposes updating the Governance to broaden the notion of
-contribution in scikit-learn and to ease subsequent related changes to
-the Governance without requiring SLEPs.
+This SLEP proposes updating the Governance to broaden the notion of contribution in scikit-learn.
+At this aim, the Technical Committee is renamed as Steering Committee, opening the
+decision process to new to be defined core roles.
```
Comment (amueller) [inline slep019/proposal.rst:20]:
```suggestion
decision process to new, to be defined, core roles.
```
Is this suggestion grammatically correct? I'm not entirely sure. Right now I find the sentence hard to parse.

File: slep019/proposal.rst [hunk 11]
```diff
@@ -13,9 +15,11 @@
  Abstract
 **********
 
-This SLEP proposes updating the Governance to broaden the notion of
-contribution in scikit-learn and to ease subsequent related changes to
-the Governance without requiring SLEPs.
+This SLEP proposes updating the Governance to broaden the notion of contribution in scikit-learn.
+At this aim, the Technical Committee is renamed as Steering Committee, opening the
```
Comment (amueller) [inline slep019/proposal.rst:19]:
```suggestion
To this end, the Technical Committee is renamed to Steering Committee, opening the
```

File: slep019/proposal.rst (hunk 7 above)
Comment (amueller) [reply slep019/proposal.rst:22]: I think it would be good to describe what the process is going to be like. "By consensus" is an option, though tends to stall, "by lazy consensus" is also an option? I don't mind not having a SLEP, but I think the SLEP should give an alternative.

Comment (cmarmo) [reply slep019/proposal.rst]: This is clarified in https://github.com/scikit-learn/enhancement_proposals/blame/master/slep019/proposal.rst#L116

File: slep019/proposal.rst [hunk 12]
```diff
@@ -85,28 +101,11 @@ Rename "Technical Committee" to "Steering Committee".
...
-at least one existing Core Contributor. The proposal must specify which
-Core Team the Contributor will be part of. The promotion is effective
-after a vote on the private Core Contributor mailing list which must
-last for two weeks and which must reach at least two-thirds positive
-majority of the cast votes.
+The current members of the Technical Committee [9]_ will ensure the transition
+for a one-years period, starting from the acceptance of this SLEP.
+During the transition period, each new created role among the "Core Contributors" is
+expected to be represented in the Steering Committee by at least one member having this
+role.
```
Comment (adrinjalali) [inline slep019/proposal.rst:108]: This is not well-defined, since it doesn't define what the transition really is.

I'm not sure if this clause is necessary. I would suggest something lightweight as:

```suggestion
The Steering Committee will accept members from any of the core contributor teams,
and after this SLEP is accepted, the definition and formation of core teams such as
the triage team and the communication team will go ahead.
```

**Review by adrinjalali — COMMENTED** [review]
Thanks @cmarmo , this looks much better now.

File: slep019/proposal.rst [hunk 13]
```diff
@@ -70,13 +74,25 @@ modifications over time.
...
+New roles, core roles, and subsequent teams will be defined.
+Core Team members will become scikit-learn "Core Contributors":
+a Contributor will be promoted to a Core Contributor after being proposed by
+at least one existing Core Contributor. The proposal must specify which
+Core Team the Contributor will be part of. The promotion will be effective
+after a vote on the private Core Contributor mailing list which must
+last for two weeks and which must reach at least two-thirds positive
+majority of the cast votes.
+
+A Core Contributor may belong to multiple core roles.
```
Comment (betatim) [inline slep019/proposal.rst:94]:
```suggestion
A Core Contributor may belong to multiple core teams.
```
nitpick to keep the wording consistent with above where "core team" is used

**Comment by cmarmo** [comment]
Hello everybody, I will be disconnected in the next days.
Please, @scikit-learn/core-devs , don't hesitate to add your comments to the current version, I will address them all when back to github. Thanks for your understanding.

File: slep019/proposal.rst (hunk 12 above)
Comment (jjerphan) [reply slep019/proposal.rst:108]: LGTM. Thank for making it lighter.

**Review by cmarmo — COMMENTED** [review]
Thank you @adrinjalali for your motivation, even if as a side effect I am forced back on line with limited capacity to act or even interact.... 😇
I have suggested some rewording as I haven't been granted the opportunity to answer previous comments... Notifying my absence apparently did not help?🤔

File: slep019/proposal.rst [hunk 14]
```diff
@@ -85,28 +101,9 @@ Rename "Technical Committee" to "Steering Committee".
...
-   -  Communication Team
-   -  Development Team
-
-A Contributor is promoted to a Core Contributor after being proposed by
-at least one existing Core Contributor. The proposal must specify which
-Core Team the Contributor will be part of. The promotion is effective
-after a vote on the private Core Contributor mailing list which must
-last for two weeks and which must reach at least two-thirds positive
-majority of the cast votes.
+The Steering Committee will accept members from any of the core contributor teams,
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply slep019/proposal.rst:104]: I'm personally not a big fan of having to integrate new members in the SC. I rather have that happen when it happens. Not sure how others feel.

File: slep019/proposal.rst [hunk 15]
```diff
@@ -85,28 +101,9 @@ Rename "Technical Committee" to "Steering Committee".
...
+and after this SLEP is accepted, the definition and formation of core teams such as
+the triage team and the communication team will go ahead.
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply slep019/proposal.rst:106]: This doesn't strictly define them, it's more of a general direction in my mind.

File: slep019/proposal.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (betatim) [reply slep019/proposal.rst:104]: IMHO the intention of these two formulations are quite different. For me "will accept" implies that "it is not illegal, but might also never happen, only time will tell", where as "will integrate" implies that "we will do this, the status quo is not enough to comply with the letter of the law". So the choice of this word has a large impact. (My word choice for how I interpret these is overly dramatic, but I couldn't think of a different one.)

The reason this difference matters to me is that the former reminds me of "commitments" that companies or governments make when they have no intentions of implementing any meaningful changes in the near term. The latter implies there will be action in the immediate future. Of course, which of the two this SLEP wants to do is up for discussion/a question of opinion. So neither is right or wrong, but they do imply different things.

[the remark this answers is not in the document]
Comment (cmarmo) [reply slep019/proposal.rst:104]: Thank you @betatim for expliciting this in a so clear way. I guess you understand why I prefer 'integrate'. No reason to put a deadline if we don't use 'integrate'. In my opinion, this SLEP lose any power of action otherwise: as I have no right to vote I will be happy to remove my name from the authors if this is merged as is.
Thanks for your understanding.

[the remark this answers is not in the document]
Comment (Micky774) [reply slep019/proposal.rst:104]: If I understand correctly, perhaps this wording is more explicit? With the above wording it seems either way to encourage a 1-1 between SC and Core Developers, whereas this allows a sub-selection.

```suggestion
The Steering Committee shall not discriminate admission on the basis of membership to specific core contributor teams.
```

File: slep019/proposal.rst (hunk 8 above)
Comment (Micky774) [reply slep019/proposal.rst:89]: I'm not sure I understand the distinction between Core "Team" vs Core "Role". Is the prior explicitly meant to refer to GitHub permissions?

Comment (cmarmo) [reply slep019/proposal.rst:89]: > Is the prior explicitly meant to refer to GitHub permissions?

Yes. Roles will exist even scikit-learn should migrate from github one day.
This is important when someone take a responsibility in the organization: github permissions should be set as this person could effectively do the job ... not the other way around... say people tasks modeled on github permissions.

**Comment by cmarmo** [comment]
Dear all,
a week has gone by.
I will try to make things as simple as possible.
I will add below two comments containing the two different possible versions of this SLEP.
Please, have a look and vote with a 👍 the one you prefer.

**Comment by cmarmo** [comment]
The Technical Committee is renamed Steering Committee.
Its composition will remain as is.
In the future, current members of the SC may nominate new members belonging to other to-be-defined core teams.

**Comment by cmarmo** [comment]
The Technical Committee is renamed Steering Committee.
Its composition will remain as is.
A deadline is set for the creation of new core roles (teams) and the integration in the SC of at least one member per core role (team).

**Comment by thomasjpfan** [comment]
For both options, I do not want the SC deciding who belongs on the SC. I prefer candidates to be nominated by **any** Core Contributor and voted on by the Core Contributors. For me, a "Core Contributors" is anyone on the existing Contributor Experience Team, Communication Team and Core Developers.

As for "at least one member per core role" from option 2, I think that would be nice to have, but I do not want to mandate it. I prefer to let the Core Teams decide if any of them wants to join the SC. If an individual wants to join, they can be nominated and voted on.

As for option 2, it proposes adding a deadline to:

1. Creating the Teams: If someone leads this effort, I am okay with adding a deadline here. Although, I think there is enough motivation to form the teams, that a deadline is not required.
2. Adding more people to the SC: I do not think a deadline helps too much here. If there are already Core Contributors interested in joining the SC, then they can get nominated and voted on when the SLEP gets accepted.

Overall, I prefer option 1 as long as the candidates get nominated by **any** Core Contributor and voted on by the Core Contributors.

**Comment by cmarmo** [comment]
> Overall, I prefer option 1 as long as the candidates get nominated by **any** Core Contributor and voted on by the Core Contributors.

Thanks @thomasjpfan I have modified option 2. following your suggestion.
Do you mind materializing your preference with a :+1: under the comment? Just to help 'counting' at the end. Thanks!

File: slep019/proposal.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (jnothman) [reply slep019/proposal.rst:103]: I find the wording of both proposals quite strange, in that I don't know what the tangible meaning of a committee accepting _or_ integrating some people is. Something I understand is processes for nomination and election, which I think is why @thomasjpfan's framing below is compelling.

A wording like:
```suggestion
The Steering Committee shall consist of members from any of the core contributor teams,
```
makes this clear, while nomination processes therefore stand as previous.

I guess the question remains of whether there is a need for affirmative action beyond the _possibility_ of inclusion. I'm not convinced that this is necessary, although I can see why we might perceive a bias towards core developers in the voting process.

**Review by jnothman — APPROVED** [review]

File: slep019/proposal.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (Micky774) [reply slep019/proposal.rst:103]: I'm also happy with @jnothman's wording suggestion

File: slep019/proposal.rst [hunk 16]
```diff
@@ -4,7 +4,8 @@
  SLEP019: Governance Update - Recognizing Contributions Beyond Code
 ####################################################################
 
-:Author: Julien Jerphanion <git@jjerphan.xyz>, Gaël Varoquaux <gael.varoquaux@normalesup.org>
+:Author: Julien Jerphanion <git@jjerphan.xyz>,
+         Gaël Varoquaux <gael.varoquaux@normalesup.org>
```
Comment (jjerphan) [inline slep019/proposal.rst:8]: For a fairer authorship.
```suggestion
:Author: Chiara Marmo <marmochiaskl@gmail.com>,
         Julien Jerphanion <git@jjerphan.xyz>,
         Gaël Varoquaux <gael.varoquaux@normalesup.org>
```

File: slep019/proposal.rst [hunk 17]
```diff
@@ -185,15 +181,24 @@ This document has been placed in the public domain [9]_.
...
-   Docs:
-   https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role
+   SLEP019: Governance Update - Recognizing Contributions Beyond Code
+   https://github.com/scikit-learn/enhancement_proposals/pull/74
 
 .. [8]
 
-   Decision Making Process, scikit-learn Governance and Decision-Making:
-   https://scikit-learn.org/dev/governance.html#decision-making-process
+   SLEP019: Governance Update - Recognizing Contributions Beyond Code
```
Comment (jjerphan) [inline slep019/proposal.rst:189]:
```suggestion
   SLEP019: Governance Update - Recognizing Contributions Beyond Code (version II)
```

**Review by jjerphan — APPROVED** [review]
Thank you for pushing this, @cmarmo.

I prefer the first option based on @thomasjpfan's comment https://github.com/scikit-learn/enhancement_proposals/pull/81#issuecomment-1349780125.

I think it's fair to add a deadline for the end of the edition of this document and the start of the vote for the SLEP.

I am approving this PR, even if I am biased to due authorship.

File: slep019/proposal.rst (hunk 16 above)
Comment (cmarmo) [reply slep019/proposal.rst:8]: @jjerphan authorship is not only a matter of reconnaissance but also of responsibility.
The preference of the core-devs is going in a direction that does not represent my opinion: I'd rather not be listed among the authors. The commit history is enough to acknowledge my participation to the edition of the SLEP.
Hence I will not commit your suggestion. Thanks for your understanding.

**Comment by glemaitre** [comment]
Regarding the discussion linked to the comments: https://github.com/scikit-learn/enhancement_proposals/pull/81#issuecomment-1345922898, https://github.com/scikit-learn/enhancement_proposals/pull/81#issuecomment-1345924934, https://github.com/scikit-learn/enhancement_proposals/pull/81#issuecomment-1349780125.

> Overall, I prefer option 1 as long as the candidates get nominated by any Core Contributor and voted on by the Core Contributors.

I agree with this statement of @thomasjpfan.

> A deadline is set for the creation of new core roles (teams) and the integration in the SC of at least one member per core role (team).

I really like the idea to have a representative and diverse SC, that must be composed of members of all "future" teams. My concern is that I have no clue how to word this and even fewer clues on how to implement it properly.

Adding quota would make sure that it happens. However, quotas are not quite ideologically aligned with voluntary-based membership: I don't want to force anyone to be in the SC if they don't want to.

So stating clearly that the SC should represent all teams would be another solution. But stating it does not mean enforcing it (as somehow brought by @betatim when discussing the difference between "accept" and "integrate").

So in the end, I don't have a revolutionary proposal to make things better and even less have the right wording to express it in a SLEP. I would therefore go along with the current proposal.

**Review by glemaitre — APPROVED** [review]

**Review by Micky774 — APPROVED** [review]
> I really like the idea to have a representative and diverse SC, that must be composed of members of all "future" teams. My concern is that I have no clue how to word this and even fewer clues on how to implement it properly.
> 
> Adding quota would make sure that it happens. However, quotas are not quite ideologically aligned with voluntary-based membership: I don't want to force anyone to be in the SC if they don't want to.
> 
> So stating clearly that the SC should represent all teams would be another solution. But stating it does not mean enforcing it (as somehow brought by @betatim when discussing the difference between "accept" and "integrate").

I agree with this sentiment. I don't think there's a reasonable way to compel this, aside from the social aspect of making a best effort of achieving that diversity of team representation. I think it makes sense to explicitly state it as the intended composition of the committee, and potentially rely on an understanding to achieve that.

I don't like how unofficial or "soft" that solution feels, but I can't think of anything stricter than that which avoids the issues brought up so far.

**Comment by cmarmo** [comment]
Dear all,

The current version of this SLEP is a vague affirmation of principles without roadmap for actions.
- "_nomination processes therefore stand as previous_" (see https://github.com/scikit-learn/enhancement_proposals/pull/81#discussion_r1047862557): without resignation from one of the current members there will be no nominations of new members.
- "_I think there is enough motivation to form the teams, that a deadline is not required._" (see https://github.com/scikit-learn/enhancement_proposals/pull/81#issuecomment-1349780125): anybody who took part to discussions in scikit-learn knows that the path from a vague consensus to the definition of the details is the most difficult and time consuming.  

The Steering Committee is likely to keep its current composition for years.
The only goal achieved by this SLEP will be a change of name... quite a small achievement for so much time spent on discussions... and no improvements in the representation of the various roles in the community.

> I don't like how unofficial or "soft" that solution feels, but I can't think of anything stricter than that which avoids the issues brought up so far.

Here the solutions adopted by [Python](https://peps.python.org/pep-0013/) itself and by [astropy](https://github.com/astropy/astropy-APEs/blob/main/APE0.rst).
- the number of the members of the SC is defined (this is not the case in scikit-learn)
- the time of service is limited (in releases or years, again, this is not the case in scikit-learn)

Therefore, before forever holding my peace, I would like to suggest the following
- definition of number of members and time of service for the SC
- a deadline for the introduction of new core roles
- the election of a new SC after the creation of those new core roles.

No quotas, nominations based on volunteering, a clear perspective of better representation.  
Do you mind considering this as a proposal?

**Comment by adrinjalali** [comment]
This whole discussion started by us wanting to add more groups and have them considered as "core contributors". I don't think anybody objects to that.

However, I don't think the sort of aggressive language, and attacks and judgements on the TC in this thread have been warranted. All TC members AFAIK agree with the sentiments proposed by the SLEP, but I don't see how the harsh language and the strict proposals for the SLEP are helpful.

This proposal is bringing in some change, and I personally very much welcome the change. However, we have 10-13 active maintainers (depending on how you count), and a TC of size 7, and some members are not necessarily very active; so there's some room for rotation there. However, we haven't cared much about how active people in the TC are since we intentionally made sure the TC doesn't have much power as long as there's a consensus between the core developers when we wrote the governance model. There has been only one instance where we had to refer to the TC, and that was a rather unimportant one. Therefore I don't see how the TC's composition would be a problem which would require a rather quick change.

To me the change in the TC/SC has the prerequisite of having more empowered and enabled people in roles other than the core developer role, and ythat is what we are going to do, and we agree on doing. But that'll take time, and once that happens, discussions about TC's composition would feel more relevant.

**Comment by cmarmo** [comment]
I forever hold my peace,then. Fell free to merge.

File: slep019/proposal.rst (hunk 15 above)
[the remark this answers is not in the document]
Comment (jjerphan) [reply slep019/proposal.rst:105]:
```suggestion
the triage team and the communication team will go ahead.
```

**Review by jjerphan — COMMENTED** [review]
A suggestion based on the last proposals from @cmarmo, @thomasjpfan, @glemaitre and @Micky774.

File: slep019/proposal.rst [hunk 18]
```diff
@@ -85,28 +100,9 @@ Rename "Technical Committee" to "Steering Committee".
...
-
-A Contributor is promoted to a Core Contributor after being proposed by
-at least one existing Core Contributor. The proposal must specify which
-Core Team the Contributor will be part of. The promotion is effective
-after a vote on the private Core Contributor mailing list which must
-last for two weeks and which must reach at least two-thirds positive
-majority of the cast votes.
+The Steering Committee shall consist of members from any of the core contributor teams,
+and after this SLEP is accepted, the definition and formation of core teams such as
+the triage team and the communication team will go ahead.
```
[the remark this answers is not in the document]
Comment (reshamas) [reply slep019/proposal.rst:105]:
```
  - has a mandate of at maximum two years
```
I can't seem to suggest edits here.
```
  - has a mandate of maximum two consecutive terms
```

[the remark this answers is not in the document]
Comment (reshamas) [reply slep019/proposal.rst:105]: I think a 2 year lifetime max is too restrictive, if that is what the current wording is implying.

[the remark this answers is not in the document]
Comment (reshamas) [reply slep019/proposal.rst:105]: Some suggestions for organization of the Steering Council:
Have some folks with 1-year terms and others with 2-year terms.

Benefits:
- gives people flexibility on how long they want to serve
- we won't have all or majority of SC members going out at the same time
- easier for inactive SC members to exit

Considerations:
- there is the consideration that there is a limit of how many SC members can be from the same institution.

Turnover can be healthy for a volunteer leadership committee:  "has a mandate of maximum two consecutive terms". However, there is the issue that there may not be enough people to serve.

[the remark this answers is not in the document]
Comment (lorentzenchr) [reply slep019/proposal.rst:105]: It like to avoid to vote too often and to limit possible candidates too much. Python with [PEP0013](https://peps.python.org/pep-0013/) has a much more powerful SC, no term limit, but a vote after each feature release, i.e. once per year. Every SC member can leave anytime.
Concerning terms, I could imagine something similar.

**Comment by lorentzenchr** [comment]
Governance is important, so is discussing it. I appreciate all the thoughts and different discussion cultures so far. I think we are getting close to finishing this SLEP, but we‘re in no rush and it needs "due diligence".

As to stricter enforcement of some of the intended changes: Even accepting a SLEP that says X will be done until some date does not guarantee that X will happen. For me, it is more important that X happens than having it in prose that it will happen. But it is important having it in prose that X is the intent and this we have (if I read correctly)😃

File: slep019/proposal.rst (hunk 18 above)
[the remark this answers is not in the document]
Comment (jnothman) [reply slep019/proposal.rst:105]: I think the words above are unclear, but the idea of explicit rather than implicit renewal makes the structure of the committee better reflect the variable nature of participation in the project. I hope that renewal will not be an opportunity to vote people out, but rather an opportunity for people to stand aside or discuss doing so.

[the remark this answers is not in the document]
Comment (jjerphan) [reply slep019/proposal.rst:105]: @jnothman: Thanks for your comment. I do not know what you are referring to with this remark:

> I think the words above are unclear.

Can you suggest a rewording of the parts which aren't clear to you? Thanks!

**Comment by jjerphan** [comment]
@lorentzenchr, thank you for your participation.

I generally agree with your point, yet I do not know what you are referring to with X in this SLEP. Can you be more specific or suggest a reword of the part which aren't clear to you? Thank you!

**Comment by jjerphan** [comment]
For your information, @thomasjpfan opened https://github.com/scikit-learn/enhancement_proposals/pull/84 to propose simplifying governance changes via SLEP020.

**Comment by adrinjalali** [comment]
Closing per https://github.com/scikit-learn/enhancement_proposals/pull/87
